config.go

```diff
@@ -270,6 +273,16 @@ func (c *WriterConfig) ConfigureWriter(config *WriterConfig) {
 		}
 	}
 
+	encodings := config.Encodings
+	if len(c.Encodings) > 0 {
```

Comment:
I am concerned that this has a side effect of modifying the input config. It can lead to unexpected behavior when reusing the config for different writers.

I also notice that we set the custom keyValueMetadata this way, so I guess it's fine to keep the same pattern.

Reply:
Hi @fpetkovski . Yes, I followed same pattern it was being done by the `KeyValueMetadata`. Let me know if this should be addressed in a different way.